TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -415,50 +415,80 @@ public class TrafficManagerTool
             DrawNodeCircle(cameraInfo, nodeId, GetToolColor(warning, false), alpha);
         }
 
+        /// <returns>the average half width of all connected segments</returns>
+        private float CalculateNodeRadius(ushort nodeId) {
+
+            float sum_half_width = 0;
+            int count = 0;
+            Constants.ServiceFactory.NetService.IterateNodeSegments(
+                nodeId,
+                (ushort segmentId, ref NetSegment segment) => {
+                    sum_half_width += segment.Info.m_halfWidth;
+                    count++;
+                    return true;
+                });
+            return sum_half_width / count;
+        }
+
         public void DrawNodeCircle(RenderManager.CameraInfo cameraInfo,
                                    ushort nodeId,
                                    Color color,
                                    bool alpha = false) {
-            NetNode[] nodesBuffer = Singleton<NetManager>.instance.m_nodes.m_buffer;
-            NetSegment segment =
-                Singleton<NetManager>.instance.m_segments.m_buffer[ nodesBuffer[nodeId].m_segment0];
-
-            Vector3 pos = nodesBuffer[nodeId].m_position;
-            float terrainY = Singleton<TerrainManager>.instance.SampleDetailHeightSmooth(pos);
-            if (terrainY > pos.y) {
-                pos.y = terrainY;
+            float r = CalculateNodeRadius(nodeId);
+            Vector3 pos = Singleton<NetManager>.instance.m_nodes.m_buffer[nodeId].m_position;
+            DrawOverlayCircle(cameraInfo, color, pos, r * 2, alpha);
+        }
+
+        public void DrawCutSegmentEnd(RenderManager.CameraInfo cameraInfo,
+                       ushort segmentId,
+                       float cut,
+                       bool bStartNode,
+                       Color color,
+                       bool alpha = false) {
+            ref NetSegment segment = ref Singleton<NetManager>.instance.m_segments.m_buffer[segmentId];
+            float width = segment.Info.m_halfWidth;
+            ushort nodeId, otherNodeId;
+
+            NetNode[] nodeBuffer = Singleton<NetManager>.instance.m_nodes.m_buffer;
+            Vector3 GetPos(ushort nodeId) {
+                Vector3 pos = nodeBuffer[nodeId].m_position;
+                float terrainY = Singleton<TerrainManager>.instance.SampleDetailHeightSmooth(pos);
+                if (terrainY > pos.y) {
+                    pos.y = terrainY;
+                }
+                return pos;
             }
 
+            bool IsMiddle(ushort nodeId) => (nodeBuffer[nodeId].m_flags & NetNode.Flags.Middle) != 0;
```

Comment:
Overlapping argument name again.

Reply:
No not this time.

Reply:
Of course that name is overlapping. Look at line 450 😉  (BTW, I have no idea where you use these variables, because I couldn't find any reference)

Reply:
Oh yes. I need to delete unused variables.